pkg/bench: benchmark latency impact of gc when hashing

dt opened the pull request:
Release note: none.
Epic: none.

dt committed:
pkg/bench: benchmark latency impact of gc when hashing

Release note: none.
Epic: none.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/116980)
<!-- Reviewable:end -->

dt commented:
```
                                                                      │  bench.txt   │
                                                                      │ max-latency  │
LatencyWhileHashing/hash=*md5.digest/block=whole/file=8.0_MiB-10        13.71k ±  1%
LatencyWhileHashing/hash=*md5.digest/block=64_KiB/file=8.0_MiB-10       1.446k ±  8%
LatencyWhileHashing/hash=*md5.digest/block=128_KiB/file=8.0_MiB-10      1.373k ±  5%
LatencyWhileHashing/hash=*md5.digest/block=512_KiB/file=8.0_MiB-10      1.121k ±  9%
LatencyWhileHashing/hash=*md5.digest/block=1.0_MiB/file=8.0_MiB-10      1.917k ±  2%
LatencyWhileHashing/hash=*md5.digest/block=2.0_MiB/file=8.0_MiB-10      3.579k ±  1%
LatencyWhileHashing/hash=*md5.digest/block=4.0_MiB/file=8.0_MiB-10      6.946k ±  1%
LatencyWhileHashing/hash=*sha256.digest/block=whole/file=8.0_MiB-10     3.933k ±  4%
LatencyWhileHashing/hash=*sha256.digest/block=64_KiB/file=8.0_MiB-10    1.383k ± 18%
LatencyWhileHashing/hash=*sha256.digest/block=128_KiB/file=8.0_MiB-10   1.370k ± 26%
LatencyWhileHashing/hash=*sha256.digest/block=512_KiB/file=8.0_MiB-10   1.266k ± 11%
LatencyWhileHashing/hash=*sha256.digest/block=1.0_MiB/file=8.0_MiB-10   1.111k ± 20%
LatencyWhileHashing/hash=*sha256.digest/block=2.0_MiB/file=8.0_MiB-10   1.177k ± 15%
LatencyWhileHashing/hash=*sha256.digest/block=4.0_MiB/file=8.0_MiB-10   2.095k ±  2%
```

dt commented:
On my GCP dev box it is much worse: 
```
name                                                                   max-latency
LatencyWhileHashing/hash=*md5.digest/block=whole/file=8.0_MiB-32        37.8k ± 9%
LatencyWhileHashing/hash=*md5.digest/block=64_KiB/file=8.0_MiB-32       8.18k ±28%
LatencyWhileHashing/hash=*md5.digest/block=128_KiB/file=8.0_MiB-32      7.72k ±14%
LatencyWhileHashing/hash=*md5.digest/block=512_KiB/file=8.0_MiB-32      9.76k ±39%
LatencyWhileHashing/hash=*md5.digest/block=1.0_MiB/file=8.0_MiB-32      10.2k ±37%
LatencyWhileHashing/hash=*md5.digest/block=2.0_MiB/file=8.0_MiB-32      15.5k ±17%
LatencyWhileHashing/hash=*md5.digest/block=4.0_MiB/file=8.0_MiB-32      21.2k ±30%
LatencyWhileHashing/hash=*sha256.digest/block=whole/file=8.0_MiB-32     45.2k ±43%
LatencyWhileHashing/hash=*sha256.digest/block=64_KiB/file=8.0_MiB-32    10.2k ±57%
LatencyWhileHashing/hash=*sha256.digest/block=128_KiB/file=8.0_MiB-32   7.89k ±21%
LatencyWhileHashing/hash=*sha256.digest/block=512_KiB/file=8.0_MiB-32   10.1k ±22%
LatencyWhileHashing/hash=*sha256.digest/block=1.0_MiB/file=8.0_MiB-32   13.9k ±21%
LatencyWhileHashing/hash=*sha256.digest/block=2.0_MiB/file=8.0_MiB-32   19.9k ±28%
LatencyWhileHashing/hash=*sha256.digest/block=4.0_MiB/file=8.0_MiB-32   30.1k ±44%
```

dt committed:
lint: allow stdlib time usage in tests

The requirement that we use timeutil to avoid per-host offsets does not apply in tests.

Release note: none.
Epic: none.

dt commented:
TFTR!

bors r+
